master/internal/db/postgres_checkpoints_intg_test.go

```diff
@@ -5,29 +5,397 @@ package db
 
 import (
 	"context"
+	"encoding/json"
+	"math"
 	"math/rand"
+	"os"
 	"sort"
+	"strconv"
 	"strings"
+	"sync"
 	"testing"
 	"time"
 	"unsafe"
 
 	"github.com/google/uuid"
 	"github.com/stretchr/testify/require"
 	"github.com/uptrace/bun"
+	"google.golang.org/protobuf/types/known/structpb"
+	"gopkg.in/yaml.v3" // Can't use ghodss/yaml since NaNs error.
 
 	"github.com/determined-ai/determined/master/pkg/etc"
 	"github.com/determined-ai/determined/master/pkg/model"
 	"github.com/determined-ai/determined/proto/pkg/checkpointv1"
+	"github.com/determined-ai/determined/proto/pkg/commonv1"
 	"github.com/determined-ai/determined/proto/pkg/modelv1"
+	"github.com/determined-ai/determined/proto/pkg/trialv1"
 )
 
+func genTrial(
+	ctx context.Context, t *testing.T, db *PgDB, exp *model.Experiment, numMetrics, numSteps int,
+) {
+	type step struct {
+		bun.BaseModel `bun:"table:steps"`
+		TrialID       int
+		TrialRunID    int
+		Metrics       map[string]any
+		TotalBatches  int
+		EndTime       time.Time
+	}
+
+	trialID := RequireMockTrial(t, db, exp).ID
+	metrics := make([]step, 0, numSteps)
+
+	endTime := time.Now()
+	for i := 0; i < numSteps; i++ {
+		m := make(map[string]float64, numMetrics)
+		for j := 0; j < numMetrics; j++ {
+			m[strconv.Itoa(j)] = rand.Float64() //nolint: gosec
+		}
+
+		metrics = append(metrics, step{
```

Comment:
question: why aren't you using AddTrainingMetrics and AddValidationMetrics here? if the goal here is to generate metrics for a trial can we make use of the functions from here: https://github.com/determined-ai/determined/pull/6222?